feature/experimental/system/gnmi/benchmarking/ate_tests/drained_configuration_convergence_time/drained_configuration_convergence_time_bgp_test.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// drained_configuration_convergence_time_test is used to verify bgp test scenarios
+// as given in gnmi1.3 testcase.
+package drained_configuration_convergence_time_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/openconfig/featureprofiles/feature/experimental/system/gnmi/benchmarking/ate_tests/internal/setup"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+const (
+	asPathRepeatValue      = 3
+	aclStatement2          = "20"
+	aclStatement3          = "30"
+	setASpathPrependPolicy = "SET-ASPATH-PREPEND"
+	setMEDPolicy           = "SET-MED"
+	bgpMED                 = 25
+)
+
+// setMED is used to configure routing policy to set BGP MED on DUT.
+func setMED(t *testing.T, dut *ondatra.DUTDevice) {
+
+	// Clear existing routing policies on DUT.
+	gnmi.Delete(t, dut, gnmi.OC().RoutingPolicy().Config())
+
+	// Configure SetMED on DUT.
+	d := &oc.Root{}
+	rp := d.GetOrCreateRoutingPolicy()
+	pdef5 := rp.GetOrCreatePolicyDefinition(setMEDPolicy)
+	actions5 := pdef5.GetOrCreateStatement(aclStatement3).GetOrCreateActions()
+	// TODO: Below code will be uncommented once configuring MED in DUT as referred in below issue is supported.
+	// Ref: https://github.com/openconfig/featureprofiles/issues/759
+	// setMedBGP := actions5.GetOrCreateBgpActions().GetOrCreateSetMed()
+	// setMedBGP.SetMed = ygot.Uint32(bgpMED)
+	actions5.GetOrCreateBgpActions().SetLocalPref = ygot.Uint32(100)
+	gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+}
+
+// setASPath is used to configure route policy set-as-path prepend on DUT.
+func setASPath(t *testing.T, dut *ondatra.DUTDevice) {
+
+	// Clear existing policies on dut.
+	gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), nil)
+
+	// Configure SetASPATH routing policy on DUT.
+	d := &oc.Root{}
```

Comment:
Since you don't use `d` other than in the subsequent line you can compact this, see suggestion below (remove `d` if adopted).